### DISCUSSION TITLE
Incorrect ratio for thumb images in the album view

### DISCUSSION THREAD

**Discussion by LudovicRousseau** [LycheeOrg/Lychee#2213] [Bugs] [answered]
### Detailed description of the problem
I have 2 Lychee web sites with the same pictures (1 test server & 1 prod server).
On 1 server the thumb images in the album view are **all** squared even if I select the 'En proportion, Grillle' 

### Steps to reproduce the issue
I don't know how to reproduce the issue.

On the server without the problem I have:
![OK](https://github.com/LycheeOrg/Lychee/assets/398496/6ea25454-2644-46a8-87e1-52f8f124558e)

On the server WITH the problem I see:
![BAD](https://github.com/LycheeOrg/Lychee/assets/398496/71dbcc82-5885-417f-bcc2-74632c7cd935)

The problem may be related to https://github.com/LycheeOrg/Lychee/issues/2196

The full size images are correct (portrait or landscape). Only the thumb images in the album view are square only.

Is it possible to force a re-generation of the thumb/small/etc images?

### Output of the diagnostics
Same as for https://github.com/LycheeOrg/Lychee/issues/2196

### Browser and system
Same as for https://github.com/LycheeOrg/Lychee/issues/2196

**Comment by ildyria** [comment]
https://github.com/LycheeOrg/Lychee/blob/master/app/Console/Commands/GenerateThumbs.php

Probably:
```sh
php artisan lychee:generate_thumbs {type : thumb name} {amount=100 : amount of photos to process} {timeout=600 : timeout time requirement}
```

**Comment by LudovicRousseau** [comment, marked as answer]
Before I regenerate the different size I had a look at the `size_variants` table in the database.
I have:
![sql](https://github.com/LycheeOrg/Lychee/assets/398496/0f5033d0-0793-430b-b05d-b34a57e16c01)
The sizes "small" and "small2" are missing.

I tried the `generate_thumbs` command but it fails:
```
# sudo -u www-data php artisan lychee:generate_thumbs {type : thumb name} {amount=100 : amount of photos to process} {timeout=600 : timeout time requirement}

  Too many arguments to "lychee:generate_thumbs" command, expected arguments   
  "type" "amount" "timeout".
```

I then used:
```
sudo -u www-data php artisan lychee:generate_thumbs small 1000 600
sudo -u www-data php artisan lychee:generate_thumbs small2x 1000 600
```
and the album view is now correct.

Maybe the missing small images could be generated when needed?
I don't know why they were not already computed.

**Comment by ildyria** [comment]
> Maybe the missing small images could be generated when needed?

This would produce some slowdown when loading a page. I am not so much in favour of that. What could be done however is add info in the diagnostics that there are X sizevariants that could be generated given the current info in the DB.

**Comment by LudovicRousseau** [comment]
Diagnostics already has a "Optimize" button for the database.
You could add a "Generate missing image sizes" as well in Diagnostics.

**Comment by ildyria** [comment]
Then I would rather add specific page for those kind of actions and get that out of diagnostics. :)

**Comment by LudovicRousseau** [comment]
Exactly.
I propose to add a page 'Maintenance" or something similar.
